core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/chunkedencoding/ChunkHeader.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.aws.chunkedencoding;
+
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+
+/**
+ * A functional interface for defining a header of a chunk.
+ * <p>
+ * The header usually depends on the chunk-data itself (hex-size), but is not required to.
+ */
+@FunctionalInterface
+@SdkProtectedApi
+public interface ChunkHeader {
+    byte[] get(byte[] chunk);
```

Comment:
It's unclear how this method is used without reading other code. Can you give an example in the javadoc and some description?